File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/stock/transaction_operation/stock_in/TransactionOperationStockInReqDTO.java
package com.example.fieldsync_inventory_api.dto.stock.transaction_operation.stock_in;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.example.fieldsync_inventory_api.dto.procurement.purchase.PurchaseRequestDTO;
import com.example.fieldsync_inventory_api.enums.OperationTask;

import java.time.Instant;
import java.util.ArrayList;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TransactionOperationStockInReqDTO {
    private Integer transactionTypeId; // Transaction type
    private OperationTask task; // define movement type for all new movement
    private Instant transactionDate;
    private Integer performedBy; // user_id
    private PurchaseRequestDTO purchase; // Could be null, for non-sale movement
    ArrayList<TOStockMovementStockInReqDTO> stockMovements; // one-two-man field like, because a transaction may contain many movement records
    private String description;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/transaction_operation/request/stock_in/TransactionOperationStockInReqDTO.kt
package com.example.fieldsync_inventory_app.data.remote.dto.transaction_operation.request.stock_in

import com.google.gson.annotations.SerializedName
import com.example.fieldsync_inventory_app.util.constans.StockInTask

data class TransactionOperationStockInReqDTO(
    val transactionTypeId: Int,
    val task: StockInTask,
    @SerializedName("performedBy") // Field name required by backend
    val performedById: Int,
    val transactionDate: String, // ISO 8601 string (none timezone)
    val description: String?,
    val purchase: TOPurchaseRequestDTO?,
    val stockMovements: List<TOStockMovementStockInReqDTO>
)